src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -441,40 +457,31 @@ private static void RemoveDocument(ImmutableArray<IIncrementalAnalyzer> analyzer
                         }
                     }
 
-                    private void ResetLogAggregatorIfNeeded(Solution currentSolution)
-                    {
-                        if (currentSolution == null || _processingSolution == null ||
-                            currentSolution.Id == _processingSolution.Id)
-                        {
-                            return;
-                        }
-
-                        SolutionCrawlerLogger.LogIncrementalAnalyzerProcessorStatistics(
-                            this.Processor._registration.CorrelationId, _processingSolution, this.Processor._logAggregator, this.Analyzers);
-
-                        this.Processor.ResetLogAggregator();
-                    }
+                   // this is only used in ResetState to find out solution has changed
+                   // and reset some states
+                    private Solution _lastSolution = null;
 
                     private async Task ResetStatesAsync()
                     {
+                        var currentSolution = this.Processor.CurrentSolution;
+                        var oldSolution = _lastSolution;
+
                         try
                         {
-                            var currentSolution = this.Processor.CurrentSolution;
-
-                            if (currentSolution == _processingSolution)
+                            if (currentSolution == oldSolution)
                             {
                                 return;
                             }
 
-                            // solution has changed
-                            ResetLogAggregatorIfNeeded(currentSolution);
+                            _lastSolution = currentSolution;
 
-                            _processingSolution = currentSolution;
+                            ResetLogAggregatorIfNeeded();
 
                             // synchronize new solution to OOP
                             await currentSolution.Workspace.SynchronizePrimaryWorkspaceAsync(currentSolution, this.CancellationToken).ConfigureAwait(false);
 
-                            await RunAnalyzersAsync(this.Analyzers, currentSolution, (a, s, c) => a.NewSolutionSnapshotAsync(s, c), this.CancellationToken).ConfigureAwait(false);
+                            // always run analyzers with latest solution
```

Comment:
1. i dont' really understand this comment.  
2. we already assigned `this.Processor.CurrentSolution` above, can we not use that?  If not, why not?

Reply:
current solution could have changed since it is assigned to local. so, I want to always use latest solution.